nx 19.2.2 cannot build repo setup for incremental builds

### Current Behavior

Instructions followed here do not work
https://nx.dev/recipes/angular/setup-incremental-builds-angular

further more I notice that there is a "tmp" folder created that seems to want to contain the built artefacts instead of the "dist" folder.

This used to work with nx 17 latest and 18 latest

### Expected Behavior

The outcome of the described docs

### GitHub Repo

https://github.com/gkamperis/nx-incremental-bug

### Steps to Reproduce

1. clone the repo
2. run `nx build my-app-m --parallel`


### Nx Report

```shell
NX   Report complete - copy this into the issue template

Node   : 22.2.0
OS     : win32-x64
npm    : 10.8.1

nx (global)        : 19.2.2
nx                 : 19.2.2
@nx/js             : 19.2.2
@nx/jest           : 19.2.2
@nx/linter         : 19.2.2
@nx/eslint         : 19.2.2
@nx/workspace      : 19.2.2
@nx/angular        : 19.2.2
@nx/devkit         : 19.2.2
@nx/eslint-plugin  : 19.2.2
@nrwl/tao          : 19.2.2
@nx/web            : 19.2.2
@nx/webpack        : 19.2.2
typescript         : 5.4.5
```


### Failure Logs

```shell
> nx run my-app-m:build:production

(node:72596) [DEP0060] DeprecationWarning: The `util._extend` API is deprecated. Please use Object.assign() instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
- Generating browser application bundles (phase: setup)...
✔ Browser application bundle generation complete.
✔ Browser application bundle generation complete.

./apps/my-app-m/src/app/app.module.ts:8:0-43 - Error: Module not found: Error: Can't resolve '@nx19/my-lib' in 'nx\nx19\apps\my-app-m\src\app'

Error: apps/my-app-m/src/app/app.component.html:1:1 - error NG8001: 'app-nx-welcome' is not a known element:
1. If 'app-nx-welcome' is an Angular component, then verify that it is part of this module.
2. If 'app-nx-welcome' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.

1 <app-nx-welcome></app-nx-welcome> <router-outlet></router-outlet>
  ~~~~~~~~~~~~~~~~

  apps/my-app-m/src/app/app.component.ts:5:16
    5   templateUrl: './app.component.html',
                     ~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component AppComponent.


Error: apps/my-app-m/src/app/app.component.html:1:35 - error NG8001: 'router-outlet' is not a known element:
1. If 'router-outlet' is an Angular component, then verify that it is part of this module.
2. If 'router-outlet' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.

1 <app-nx-welcome></app-nx-welcome> <router-outlet></router-outlet>
                                    ~~~~~~~~~~~~~~~

  apps/my-app-m/src/app/app.component.ts:5:16
    5   templateUrl: './app.component.html',
                     ~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component AppComponent.


Error: apps/my-app-m/src/app/app.module.ts:7:29 - error TS2307: Cannot find module '@nx19/my-lib' or its corresponding type declarations.

7 import { MyLibModule } from '@nx19/my-lib';
                              ~~~~~~~~~~~~~~


Error: apps/my-app-m/src/app/app.module.ts:11:12 - error NG1010: Value

——————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 NX   Ran target build for project my-app-m (7s)

   ×  1/1 failed
   √  0/1 succeeded [0 read from cache]
```


### Package Manager Version

_No response_

### Operating System

- [ ] macOS
- [ ] Linux
- [X] Windows
- [ ] Other (Please specify)

### Additional Information

_No response_

--- a/docs/shared/guides/setup-incremental-builds-angular.md
+++ b/docs/shared/guides/setup-incremental-builds-angular.md
@@ -61,6 +61,7 @@ executor to `@nx/angular:dev-server` as shown below:
   ...
   "targets": {
     "build": {
+      "dependsOn": ["^build"],
       "executor": "@nx/angular:webpack-browser",
       "outputs": [
         "{options.outputPath}"
@@ -90,6 +91,19 @@ executor to `@nx/angular:dev-server` as shown below:
   }
 },
 ```
+
+{% callout type="note" title="Add Executor to Target Defaults" %}
+If you'd like to avoid adding `"dependsOn": ["^build"]` to every application in your workspace that uses `@nx/angular:webpack-browser` you can add it to the `"targetDefaults"` section of the `nx.json`:
+
+```json
+"targetDefaults": {
+  "@nx/angular:webpack-browser": {
+    "dependsOn": ["^build"]
+  }
+}
+```
+
+{% /callout %}
 
 ## Running and serving incremental builds
 

--- a/packages/angular/src/generators/setup-mf/lib/change-build-target.ts
+++ b/packages/angular/src/generators/setup-mf/lib/change-build-target.ts
@@ -5,6 +5,7 @@ import {
   readProjectConfiguration,
   updateProjectConfiguration,
 } from '@nx/devkit';
+import { addBuildTargetDefaults } from '@nx/devkit/src/generators/target-defaults-utils';
 
 export function changeBuildTarget(host: Tree, options: Schema) {
   const appConfig = readProjectConfiguration(host, options.appName);
@@ -27,4 +28,6 @@ export function changeBuildTarget(host: Tree, options: Schema) {
   };
 
   updateProjectConfiguration(host, options.appName, appConfig);
+
+  addBuildTargetDefaults(host, '@nx/angular:webpack-browser');
 }

